homeassistant/components/alphaess/sensor.py

```diff
@@ -0,0 +1,127 @@
+"""Alpha ESS Sensor definitions."""
+import datetime
+import logging
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+)
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_ENERGY,
+    ENERGY_KILO_WATT_HOUR,
+    PERCENTAGE,
+)
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTR_ENTRY_TYPE, DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    for invertor in coordinator.data:
+        serial = invertor["sys_sn"]
+        async_add_entities(
+            [
+                AlphaESSSensor(coordinator, entry, serial, "Solar Production"),
+                AlphaESSSensor(coordinator, entry, serial, "Solar to Battery"),
+                AlphaESSSensor(coordinator, entry, serial, "Solar to Grid"),
+                AlphaESSSensor(coordinator, entry, serial, "Solar to Load"),
+                AlphaESSSensor(coordinator, entry, serial, "Total Load"),
+                AlphaESSSensor(coordinator, entry, serial, "Grid to Load"),
+                AlphaESSSensor(coordinator, entry, serial, "Grid to Battery"),
+                AlphaESSSensor(coordinator, entry, serial, "State of Charge"),
+                AlphaESSSensor(coordinator, entry, serial, "Charge"),
+                AlphaESSSensor(coordinator, entry, serial, "Discharge"),
+            ]
+        )
+
+    return
+
+
+class AlphaESSSensor(CoordinatorEntity, SensorEntity):
+    """Alpha ESS Base Sensor."""
+
+    def __init__(self, coordinator, config, serial, name):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._config = config
+        self._name = name
+        self._serial = serial
+        self._coordinator = coordinator
+
+        for invertor in self._coordinator.data:
+            serial = invertor["sys_sn"]
+            if self._serial == serial:
+                model = invertor["minv"]
+
+        self._attr_device_info = {
+            ATTR_IDENTIFIERS: {(DOMAIN, serial)},
+            ATTR_NAME: f"Alpha ESS Energy Statistics : {serial}",
+            ATTR_MANUFACTURER: "AlphaESS",
+            ATTR_MODEL: model,
+            ATTR_ENTRY_TYPE: DeviceEntryType.SERVICE,
+        }
```

Comment:
```suggestion
        self._attr_device_info = DeviceInfo(
            entry_type=DeviceEntryType.SERVICE,
            identifiers={(DOMAIN, serial)},
            manufacturer="AlphaESS",
            model=model,
            name= f"Alpha ESS Energy Statistics : {serial}",
        )
```

Use the typed device info from `from homeassistant.helpers.entity import DeviceInfo`

Reply:
Thanks for the advice!  Much appreciated.